lean4-infoview/src/infoview/interactiveCode.tsx

```diff
@@ -47,6 +55,8 @@ export function InteractiveTaggedText<T>({ fmt, InnerTagUi }: InteractiveTaggedT
     else throw new Error(`malformed 'TaggedText': '${fmt}'`)
 }
 
+export const MemoedInteractiveTaggedText = genericMemo(InteractiveTaggedText)
```

Comment:
The main reason is that the suggestion at https://stackoverflow.com/a/70890101 used this style. The other reason is that `InteractiveTaggedText` is also used in the trace explorer and I did not want to bother with testing whether the trace explorer still works correctly after this memoization, so it still uses the unmemoized variant.

Reply:
> I did not want to bother with testing whether the trace explorer still works correctly after this memoization

I am 99% certain it will work correctly; memoization should not change the observable behaviour of a functional component. It would break usages wherein a prop `p` is passed, then internally mutated, then passed again as `p` which is still shallowly equal. In that case the memoized component should re-render, but won't. But iirc we don't do that kind of thing here - reference equality implies deep equality - so may as well `memo` everything.